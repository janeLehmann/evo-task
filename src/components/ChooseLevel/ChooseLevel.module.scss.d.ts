declare const styles = {
  chooseLevel: string,
  chooseLevelTitle: string,
  chooseLevelDesc: string,
  chooseLevelList: string,
  chooseLevelItem: string,
  chooseLevelItemButton: string,
  chooseLevelItemText: string,
};

export default styles;
