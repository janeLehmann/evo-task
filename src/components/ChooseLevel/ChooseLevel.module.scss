@import 'src/static/styles/_mixins.scss';

.chooseLevel {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
}

.chooseLevelTitle {
  text-align: center;
  @include medium();
  margin-bottom: 20px;
  font-size: 40px;
}

.chooseLevelDesc {
  text-align: center;
  @include regular();
  margin-bottom: 30px;
  font-size: 25px;
}

.chooseLevelList {
  display: flex;
  align-items: center;
  justify-content: center;
}

.chooseLevelItem {
  width: 120px;
  margin-right: 40px;

  &:last-child {
    margin-right: 0;
  }
}

.chooseLevelItemButton {
  @include reset-button-styles();
  width: 100%;
  height: 40px;
  background: #C0C0C0;
  border-left: 4px solid #808080;
  border-top: 4px solid #808080;
  margin-bottom: 20px;
  font-size: 20px;
  @include medium();
  transition: background .3s ease-in-out;

  &:hover {
    background: #7a7a7a;
  }

  .chooseLevelItem:first-child & {
    color: #0000ff;
  }

  .chooseLevelItem:nth-child(2) & {
    color: #008000;
  }

  .chooseLevelItem:nth-child(3) & {
    color: #ff1821;
  }

  .chooseLevelItem:nth-child(4) & {
    color: #a6151c;
  }
}

.chooseLevelItemText {
  text-align: center;
  font-size: 20px;
  @include regular();
}


@media screen and (max-width: 700px) {
  .chooseLevelList {
    display: block;
    width: 100%;
    padding-left: 20px;
    padding-right: 20px;
  }

  .chooseLevelItem {
    width: 100%;
    margin-right: 0;
    margin-bottom: 20px;

    &:last-child {
      margin-bottom: 0;
    }
  }

  .chooseLevelItemButton {
    height: 60px;
  }
}
