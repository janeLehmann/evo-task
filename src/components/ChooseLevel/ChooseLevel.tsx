import React from 'react';
import styles from './ChooseLevel.module.scss';

interface Level {
  title: string;
  desc: string;
  onClick: any;
}

interface ChooseLevelProps {
  levels: Level[];
}

const ChooseLevel: React.FC<ChooseLevelProps> = ({ levels }) => {
  return (
    <div className={styles.chooseLevel}>
      <h1 className={styles.chooseLevelTitle}>Choose the level</h1>
      <div className={styles.chooseLevelDesc}>
        After choosing the level the game will start automatically
      </div>

      <div className={styles.chooseLevelList}>
        {levels.map((item) => (
          <div className={styles.chooseLevelItem} key={item.title} data-testid="choose-level-item">
            <button className={styles.chooseLevelItemButton} type="button" onClick={item.onClick}>
              Level {item.title}
            </button>
            <div className={styles.chooseLevelItemText}>{item.desc}</div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default ChooseLevel;
