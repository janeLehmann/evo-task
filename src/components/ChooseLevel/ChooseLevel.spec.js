import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { toHaveClass, toBeInTheDocument, toHaveTextContent } from '@testing-library/jest-dom';
import { create } from 'react-test-renderer';

import ChooseLevel from './ChooseLevel';

expect.extend({ toHaveClass, toBeInTheDocument, toHaveTextContent });

const levels = [
  {
    title: '1',
    desc: '10x10 cells',
    onClick: jest.fn(),
  },
  {
    title: '2',
    desc: '20x40 cells',
    onClick: jest.fn(),
  },
];

const renderComponent = (overrides = {}, useReactTestRenderer = false) => {
  const props = {
    levels: levels,
    ...overrides,
  };

  if (!useReactTestRenderer) {
    /* eslint-disable react/prop-types, react/jsx-props-no-spreading */
    return render(<ChooseLevel {...props} />);
    /* eslint-enable react/prop-types, react/jsx-props-no-spreading */
  }

  /* eslint-disable react/prop-types, react/jsx-props-no-spreading */
  return create(<ChooseLevel {...props} />);
  /* eslint-enable react/prop-types, react/jsx-props-no-spreading */
};

it('should render list of 2 items', () => {
  const buttonClick1 = jest.fn();
  const buttonClick2 = jest.fn();
  const $ = renderComponent({
    levels: [
      {
        ...levels[0],
        onClick: buttonClick1,
      },
      {
        ...levels[1],
        onClick: buttonClick2,
      },
    ],
  });

  const items = $.getAllByTestId('choose-level-item');
  expect(items.length).toBe(2);
  const button1 = $.getByText('Level 1');
  const button2 = $.getByText('Level 2');
  expect(button1).toBeInTheDocument();
  expect(button2).toBeInTheDocument();

  fireEvent.click(button1);
  expect(buttonClick1).toHaveBeenCalledTimes(1);

  fireEvent.click(button2);
  expect(buttonClick2).toHaveBeenCalledTimes(1);

  expect($.container).toMatchSnapshot();
});
