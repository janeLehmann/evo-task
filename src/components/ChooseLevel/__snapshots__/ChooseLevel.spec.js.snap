// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of 2 items 1`] = `
<div>
  <div
    class="chooseLevel"
  >
    <h1
      class="chooseLevelTitle"
    >
      Choose the level
    </h1>
    <div
      class="chooseLevelDesc"
    >
      After choosing the level the game will start automatically
    </div>
    <div
      class="chooseLevelList"
    >
      <div
        class="chooseLevelItem"
        data-testid="choose-level-item"
      >
        <button
          class="chooseLevelItemButton"
          type="button"
        >
          Level 
          1
        </button>
        <div
          class="chooseLevelItemText"
        >
          10x10 cells
        </div>
      </div>
      <div
        class="chooseLevelItem"
        data-testid="choose-level-item"
      >
        <button
          class="chooseLevelItemButton"
          type="button"
        >
          Level 
          2
        </button>
        <div
          class="chooseLevelItemText"
        >
          20x40 cells
        </div>
      </div>
    </div>
  </div>
</div>
`;
